### DISCUSSION TITLE
snapshot and containerd, could you please elaborate?

### DISCUSSION THREAD

**Discussion by rthamrin** [k3s-io/k3s#5440] [Q&A]
I installed k3s inside of my machine and install the Nginx ingress controller load balancer.
I am confused about the function of containerd.snapshooter.***** , I don't get it what does it means. How exactly if I wanna do something inside of my Nginx folder or file ? what does snapshot mean here, and any idea where I can find my original file. because is not there ```/opt/bitnami/nginx/nginx.conf``` , feels my files are trapped inside of this snapshoot. I tried to figure it out because my folder is supposed to be in /opt/bitnami/nginx or /usr/share/nginx not inside of these snapshots

```
/var/lib/docker/overlay2/b130655400dfb35828ea63b5343c6d65fc7d487dae19e33e3ede9a9952986aa5/diff/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/6676/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/3695/fs/opt/bitnami/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/3335/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7177/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/8186/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/2869/fs/opt/bitnami/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/3328/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7518/fs/opt/bitnami/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7832/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/1954/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/8201/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/1962/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7939/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7312/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/6654/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7193/fs/usr/share/nginx/html
/var/lib/rancher/k3s/agent/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/7801/fs/usr/share/nginx/html
```

**Comment by brandond** [comment]
You're asking questions about the internals of the containerd overlay filesystem. These are implementation-specific bits of how containerd (the k3s container runtime) stores the layers of an image or running container. If you're not familiar with how containers work or how to get at files within a running container, poking around in the filesystem is probably not a good idea.

What specifically are you trying to accomplish? What do you need to do with that file?

**Reply by rthamrin** [reply]
I want to modify my Nginx.conf file by running configmap.yaml file to improve my Nginx ingress load balancer.

**Reply by brandond** [reply]
I'm not sure what you mean by "running configmap.yaml". Are there some instructions that your following? It sounds like you want to use `kubectl edit configmap` to edit one of the ingress-nginx configmaps, or edit the chart values to do so if using the helm chart. You wouldn't edit that file directly in the running container; thats not how things are done in Kubernetes.

**Reply by rthamrin** [reply]
this is what I am going to do ```https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/``` 

and

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: nginx-conf
  namespace: nginx-ingress
data:
  nginx.conf: |-
    user nginx;
    worker_processes  10;
    events {
      worker_connections  10240;
    }
    http {
      server {
          listen       80;
          server_name  localhost;
          location / {
            root   /usr/share/nginx/html; #Change this line
            index  index.html index.htm;
        }
      }
    }
```
i want to point my folder ``` root   /usr/share/nginx/html; #Change this line```

**Reply by brandond** [reply]
OK, so the appropriate way to set those would be by editing the configmap in the cluster using kubectl, not by editing the copy of the configmap that's been projected down into the pod.

**Reply by rthamrin** [reply]
basically , I just use kubectl edit configmap ? but how if i want to edit the template which is i need to mount spesific folder:

like this ```https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/custom-template/``` i need to mount the ```/etc/nginx/template```
```
 volumeMounts:   
  - mountPath: /etc/nginx/template
            name: nginx-template-volume
            readOnly: true
      volumes:
        - name: nginx-template-volume
          configMap:
            name: nginx-template
            items:
            - key: nginx.tmpl
              path: nginx.tmpl
````
